src/main/resources/assets/computercraft/lua/rom/programs/list.lua

```diff
@@ -31,8 +50,33 @@ end
 table.sort( tDirs )
 table.sort( tFiles )
 
-if term.isColour() then
-    textutils.pagedTabulate( colors.green, tDirs, colors.white, tFiles )
+local function writeFileTable( tPrint, tList )
+    for i, v in ipairs( tList ) do
+        tAttr = fs.attributes( shell.resolve( v ) )
+        tSingeEntry = {v}
+        if tAttr["isDir"] then
+            table.insert( tSingeEntry, "Dir" )
+        else
+            table.insert( tSingeEntry, "File" )
+        end
+        sDate = settings.get("list.date_format")
+        table.insert( tSingeEntry, tostring( tAttr["size"] ) )
+        table.insert( tSingeEntry, os.date( sDate, tAttr["created"] ) )
+        table.insert( tSingeEntry, os.date( sDate, tAttr["modification"] ) )
+        table.insert( tPrint, tSingeEntry )
+    end
+    return tPrint
+end
+
+if bShowAttr then
+    tPrint = {{"Name","Type","Size","Created","Modified"}}
```

Comment:
I'd leave off the "Created" column - as you've mentioned elsewhere, there's limited space, and IMO we don't need to include both. This'd mean we can use `%Y-%m-%d` instead of `%y`.

Ideally we could also leave off "Type" and just colour "Name", but I guess `tabulate` doesn't allow that.